api/apps/articles/model/save.coffee

```diff
@@ -177,7 +177,7 @@ getPartnerLink = (artwork) ->
     return cb err if err
     publishing = (input.published and not article.published) or (input.scheduled_publish_at and not article.published)
     article = _.extend article, _.omit(input, 'sections'), updated_at: new Date
-    article.sections = [] unless input.sections?.length
+    article.sections = [] unless input.sections.length > 0
```

Comment:
I think you still want the `?` guard here in case a client doesn't send `sections`.